scripts/export_event_types.py

```diff
@@ -45,71 +77,119 @@
-        dtf["event_type"] = -99
-        # from IPython import embed
-        # embed()
-        for energy_key in dtf_e_test.keys():
-            if "gamma_cone" in dl2_file:
+            # Calculate event types for proton and electron events, using the same event type
+            # thresholds as in the gammas:
+            if offset_partition:
+                d_types = event_types.partition_event_types_offsetwise(
+                    dtf_test,
+                    labels=labels,
+                    log_e_bins=event_type_log_e_bins,
+                    offset_bins=event_type_offset_bins,
+                    n_types=n_types,
+                    event_type_bins=event_type_partition,
+                )
+            else:
+                d_types = event_types.partition_event_types(
+                    dtf_test,
+                    labels=labels,
+                    log_e_bins=event_type_log_e_bins,
+                    n_types=n_types,
+                    event_type_bins=event_type_partition,
+                )
+        # Start creating the event_type column within the original dataframe,
+        # setting the default value: -3.
+        # There shouldn't be many -3s at the end. If that's the case, it must be understood.
+        dtf["event_type"] = -3
+
+        if particle is gamma:
+            # Assign type -1 to the events used for the training.
+            for energy_key in dtf_e_train.keys():
                 dtf.loc[dtf_e_train[energy_key].index.values, "event_type"] = -1
-        dtf.loc[dtf_test["default"].index.values, "event_type"] = dtf_test["default"]["event_type"]
+
+        dtf.loc[dtf_test[suffix].index.values, "event_type"] = dtf_test[suffix]["event_type"]
 
         print("A total of {} events will be written.".format(len(dtf["event_type"])))
-        for event_type in [-99, 1, 2, 3]:
+        dtf_7 = dtf[dtf["cut_class"] != 7]
+        for event_type in [-3, -2, -1, 1, 2, 3]:
             print(
                 "A total of {} events of type {}".format(
                     np.sum(dtf["event_type"] == event_type), event_type
                 )
             )
-
-        with open(dl2_file.replace(".root", ".txt"), "w") as txt_file:
+            print(
+                "A total of {} events of type {} for gamma-like events".format(
+                    np.sum(dtf_7["event_type"] == event_type), event_type
+                )
+            )
+        # Summary:
+        # Types 1, 2 and 3 are actual reconstructed event types, where 1 is the best type and
+        # all three should have similar statistics.
+        # Type -1 is assigned to the train sample.
+        # Type -2 is the default value when assigning event types to the test sample in
+        # partition_event_types_2. As that function works for each bin of energy and offset,
+        # events outside these bins will have type -2.
+        # Type -3 is the default value for the complete tables of each particle.
```

Comment:
Do you still expect to have type -2 and -3 after partitioning? I thought that if all went well you shouldn't have any of them.

Reply:
You are right, there shouldn't be any, but there still are...
-2 type is understood (I think), as we are providing the bins for the energy and offset partition, some events lie outside those bins, so they don't get an event type and remain -2. This should be reduced almost to zero if we add a last bigger offset bin that includes everything above the previous bin, so I'll do it as you suggest.
-3 type is a mystery for me yet. There are very few events that get this type, but they still exist...

Reply:
OK, so let's keep this description until both issues are resolved.